### ISSUE TITLE
[Bug] [UWP] Item Spacing still exists even when ItemSpacing is 0

### ISSUE THREAD

**Issue by Happypig375** [xamarin/Xamarin.Forms#7842]
### Description
Item Spacing still exists even when ItemSpacing is 0

### Steps to Reproduce

```xaml
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             xmlns:local="clr-namespace:WhereIsImage"
             mc:Ignorable="d"
             x:Class="WhereIsImage.XF">
    <CollectionView BackgroundColor="Blue">
        <CollectionView.ItemsSource>
            <x:Array Type="{x:Type x:String}">
                <x:String>Baboon</x:String>
                <x:String>Capuchin Monkey</x:String>
                <x:String>Blue Monkey</x:String>
                <x:String>Squirrel Monkey</x:String>
                <x:String>Golden Lion Tamarin</x:String>
                <x:String>Howler Monkey</x:String>
                <x:String>Japanese Macaque</x:String>
            </x:Array>
        </CollectionView.ItemsSource>
        <CollectionView.ItemsLayout>
            <GridItemsLayout Orientation="Vertical" Span="1" HorizontalItemSpacing="0" VerticalItemSpacing="0" />
        </CollectionView.ItemsLayout>
        <CollectionView.ItemTemplate>
            <DataTemplate>
                <Label Text="{Binding}" BackgroundColor="AliceBlue"/>
            </DataTemplate>
        </CollectionView.ItemTemplate>
    </CollectionView>
</ContentPage>
```

### Expected Behavior
No spacing exists between items

### Actual Behavior
Spacing exists

### Basic Information

- Version with issue: 4.2
- Last known good version: Idk
- IDE: VS16.3.2
- Platform Target Frameworks: 
  - UWP:  17134
- Affected Devices: My PC

### Screenshots

![image](https://user-images.githubusercontent.com/19922066/66270888-f2507200-e88a-11e9-838e-8353bc8e2612.png)

### Reproduction Link
[WhereIsImage2.zip](https://github.com/xamarin/Xamarin.Forms/files/3694504/WhereIsImage2.zip)

**Comment by jfversluis** [comment]
Also making the item spacing bigger seems to have no effect. Looks like this isn't built yet or doesn't work properly.

cc: @jsuarezruiz

**Comment by SteveMaz** [comment]
LinearItemsLayout also has the same problem.

**Comment by velocitysystems** [comment]
We too can confirm default spacing is applied on UWP. This needs to be implemented in Windows to bring consistency of experience along with the iOS and Android renderers.

**Comment by velocitysystems** [comment]
@jfversluis @jsuarezruiz Fixed with #8839.

**Comment by jfversluis** [comment]
Great work @velocitysystems! Thanks for the effort :)

**Comment by samhouts** [comment]
closed by #8839
